framework/metricsresource/resource_test.go

```diff
@@ -0,0 +1,136 @@
+package metricsresource
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/giantswarm/operatorkit/framework"
+)
+
+// Test_MetricsResource_ProcessCreate_ResourceOrder ensures the resource's
+// methods are executed as expected when creating resources using the wrapping
+// prometheus resource.
+func Test_MetricsResource_ProcessCreate_ResourceOrder(t *testing.T) {
+	tr := &testResource{}
+	rs := []framework.Resource{
+		tr,
+	}
+
+	config := DefaultWrapConfig()
+	config.Namespace = t.Name()
```

Comment:
As noted in another comment the combination between namespace and subsystem has to be unique or the Prometheus vector registration panics. `testing.T.Name` returns the function name, which is unique, so the tests do not panic.